system/security/shared/SecureUser.hpp

```diff
@@ -328,6 +328,11 @@ class CSecureUser : implements ISecUser, implements ISecCredentials, public CInt
     {
         return false;
     }
+
+    virtual bool isCanonicalMatch(const char* name) const override
+    {
+        return (name && strieq(m_name, name));
```

Comment:
m_name is never tested to ensure it isn't null. Not sure if that's an exposure here, but consider adding a check in the ctor above, or at least a check here for n_name not null or empty

Reply:
Either way is fine, but m_name.str() will be easier to read.